packages/foam-vscode/src/features/filtered-resources.ts

```diff
@@ -0,0 +1,252 @@
+      item.tooltip = getNoteTooltip(content);
+    }
+    return item;
+  }
+
+  private computeFilteredResources(): void {
+    this.filteredResources = this.workspace
+      .list()
+      .filter(this.filterPredicate)
+      .filter(resource => !this.isMatch(resource.uri))
+      .sort(this.sort);
+  }
+
+  private isMatch(uri: URI) {
+    return micromatch.isMatch(uri.fsPath, this.exclude);
+  }
+
+  private getGlobs(fsURI: URI[], globs: string[]): string[] {
+    globs = globs.map(glob => (glob.startsWith('/') ? glob.slice(1) : glob));
+
+    const exclude: string[] = [];
+
+    for (const fsPath of fsURI) {
+      let folder = fsPath.path.replace(/\\/g, '/');
+      if (folder.substr(-1) === '/') {
+        folder = folder.slice(0, -1);
+      }
+      exclude.push(...globs.map(g => `${folder}/${g}`));
+    }
+
+    return exclude;
+  }
+
+  private getFilteredResourcesByDirectory(): FilteredResourcesByDirectory {
+    const filtered: FilteredResourcesByDirectory = {};
+    for (const resource of this.filteredResources) {
+      const p = resource.uri.path.replace(this.root, '');
+      const { dir } = path.parse(p);
+
+      if (filtered[dir]) {
+        filtered[dir].push(resource);
+      } else {
+        filtered[dir] = [resource];
+      }
+    }
+
+    for (const k in filtered) {
+      filtered[k].sort(this.sort);
+    }
+
+    return filtered;
+  }
+
+  private sort(a: Resource, b: Resource) {
+    const titleA = getTitle(a);
+    const titleB = getTitle(b);
+    return titleA.localeCompare(titleB);
+  }
+}
+
+export interface FilteredResourcesProviderConfig
+  extends FilteredResourcesConfig {
+  workspacesURIs: URI[];
+  includeLinks?: boolean;
+}
+
+type FilteredResourcesByDirectory = { [key: string]: Resource[] };
+
+type FilteredResourceTreeItem = FilteredResource | Directory;
+
+class FilteredResource extends vscode.TreeItem {
+  constructor(public readonly resource: Resource, contextValue: string) {
+    super(getTitle(resource), vscode.TreeItemCollapsibleState.None);
+    this.description = resource.uri.path;
+    this.tooltip = undefined;
+    if (isPlaceholder(resource)) {
+      this.command = {
+        command: 'foam-vscode.open-placeholder-note',
+        title: 'Foam: Open Placeholder Note',
+        arguments: [resource.uri],
+      };
+    } else {
+      this.command = {
+        command: 'vscode.open',
+        title: 'Open File',
+        arguments: [resource.uri],
+      };
+    }
+    this.contextValue = contextValue;
+
+    let iconStr: string;
+    if (isAttachment(this.resource)) {
+      iconStr = 'file-media';
+    } else if (isPlaceholder(this.resource)) {
+      iconStr = 'new-file';
+    } else {
+      iconStr = 'note';
+    }
+    this.iconPath = new vscode.ThemeIcon(iconStr);
+  }
+}
+
+export class Directory extends vscode.TreeItem {
+  constructor(
+    public readonly dir: string,
+    public readonly resources: Resource[],
+    contextValue: string
```

Comment:
this is not the `contextValue`, but the type of resource which is used for the label, we should call this `itemLabel` or something like that